### PR TITLE
drivers: sensor: bme68x_iaq: Add support for SPI bus

### PR DESCRIPTION
**Pull request by purpl3F0x**
Adds support for a bme680x sensor SPI bus.

### CODE REVIEW

**Comment by purpl3F0x** [comment]
Should I include a commit message body due avoid the compliance error ? title is pretty descriptive of the commit.

**Comment by maxd-nordic** [comment]
> Should I include a commit message body due avoid the compliance error ? title is pretty descriptive of the commit.

Yes, the compliance rules require that. Sorry about the hassle! You could add something like "This patch adds SPI bus support for the bme68x_iaq driver."

**Comment by purpl3F0x** [comment]
> Yes, the compliance rules require that. Sorry about the hassle! You could add something like "This patch adds SPI bus support for the bme68x_iaq driver."

Νo problem, wasn't sure if it will be a blocking matter. Hopefully it's okay now.

File: drivers/sensor/bme68x_iaq/bme68x_iaq.c [hunk 1]
```diff
@@ -111,18 +113,62 @@ static void state_save(const struct device *dev)
...
+	return spi_write_dt(&bme68x_spi_spec, &tx);
+
+#elif BME68x_BUS_I2C
 	uint8_t buf[len + 1];
 
 	buf[0] = reg_addr;
 	memcpy(&buf[1], reg_data_ptr, len);
 
-	return i2c_write_dt(&bme68x_i2c_spec, buf, ARRAY_SIZE(buf));
+	return i2c_write_dt(&bme68x_i2c_specc, buf, ARRAY_SIZE(buf));
```
Comment (maxd-nordic) [inline drivers/sensor/bme68x_iaq/bme68x_iaq.c:141]:
```suggestion
	return i2c_write_dt(&bme68x_i2c_spec, buf, ARRAY_SIZE(buf));

```

File: drivers/sensor/bme68x_iaq/bme68x_iaq.c [hunk 2]
```diff
@@ -401,12 +450,20 @@ static int bme68x_bsec_init(const struct device *dev)
...
 
-	if (!device_is_ready(bme68x_i2c_spec.bus)) {
+#if BME68x_BUS_SPI
+	if (!spi_is_ready_dt(&bme68x_spi_spec)) {
+		LOG_ERR("SPI device not ready");
+		return -ENODEV;
+	}
+	data->dev.intf = BME68X_SPI_INTF;
+#elif BME68x_BUS_I2C
+	if (!device_is_ready(bme68x_i2c_spec)) {
```
Comment (maxd-nordic) [inline drivers/sensor/bme68x_iaq/bme68x_iaq.c:460]:
```suggestion
	if (!device_is_ready(bme68x_i2c_spec.bus)) {
```

File: drivers/sensor/bme68x_iaq/bme68x_iaq.h [hunk 3]
```diff
@@ -7,13 +7,34 @@
...
+#define DT_DRV_COMPAT bosch_bme680
+
+#define BME68x_BUS_SPI DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#define BME68x_BUS_I2C DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+
+
+#if BME68x_BUS_SPI
+	#include <zephyr/drivers/spi.h>
+#elif BME68x_BUS_I2C
+	#include <drivers/i2c.h>
```
Comment (maxd-nordic) [inline drivers/sensor/bme68x_iaq/bme68x_iaq.h:28]:
```suggestion
	#include <zephyr/drivers/i2c.h>
```

**Review by maxd-nordic — CHANGES_REQUESTED** [review]
With the suggested changes, the i2c build works again.

File: drivers/sensor/bme68x_iaq/bme68x_iaq.c (hunk 1 above)
Comment (purpl3F0x) [reply drivers/sensor/bme68x_iaq/bme68x_iaq.c]: oops

**Comment by purpl3F0x** [comment]
Sorry for the typos.

Btw, somewhat unrelated to the commit, should support for other BSEC metrics (VOC, CO2 etc.) be added?

**Comment by maxd-nordic** [comment]
> Sorry for the typos.
> 
> Btw, somewhat unrelated to the commit, should support for other BSEC metrics (VOC, CO2 etc.) be added?

Oh, if you know how to add these, this could be pretty cool! Last time I checked it looked a bit involved to get other metrics.

For now, it would be good to add a build test for the SPI case as well. For this, we would add the following file at `nrf/samples/sensor/bme68x_iaq/boards/nrf9160dk_nrf9160_ns.overlay`:

```
/*
 * Copyright (c) 2023 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 */
&arduino_spi {
	status = "okay";

        bme680@0 {
                compatible = "bosch,bme680";
                reg = <0x0 >;
		spi-max-frequency = <DT_FREQ_M(8)>;
        };
};
```
And adding this to `nrf/samples/sensor/bme68x_iaq/sample.yaml`:
```
  sample.sensor.bme68x.spi:
    depends_on: spi
    harness: sensor
    integration_platforms:
      - nrf9160dk_nrf9160_ns
    platform_allow: >-
      nrf9160dk_nrf9160_ns
    tags: sensors
```

**Comment by purpl3F0x** [comment]
> sample.sensor.bme68x.spi:
>     depends_on: spi
>     harness: sensor
>     integration_platforms:
>       - nrf9160dk_nrf9160_ns
>     platform_allow: >-
>       nrf9160dk_nrf9160_ns
>     tags: sensors

Added the suggested additions, I also renamed the first test to `sample.sensor.bme68x.i2c` to be more descriptive.

File: samples/sensor/bme68x_iaq/boards/nrf9160dk_nrf9160_ns.overlay [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ */
+ &arduino_spi {
+	status = "okay";
+
+        bme680@0 {
+                compatible = "bosch,bme680";
+                reg = <0x0 >;
+		spi-max-frequency = <DT_FREQ_M(8)>;
+        };
+};
```
Comment (maxd-nordic) [inline samples/sensor/bme68x_iaq/boards/nrf9160dk_nrf9160_ns.overlay:14]:
```suggestion
};

```

File: samples/sensor/bme68x_iaq/sample.yaml [hunk 5]
```diff
@@ -35,3 +35,11 @@ tests:
       ordered: true
       regex:
         - "(\\d+\\.\\d+); press: (\\d+\\.\\d+); humidity: (\\d+\\.\\d+); iaq: (\\d+)"
+    sample.sensor.bme68x.spi:
```
Comment (maxd-nordic) [inline samples/sensor/bme68x_iaq/sample.yaml:38]:
```suggestion
  sample.sensor.bme68x.spi:
```

**Review by maxd-nordic — APPROVED** [review]

File: samples/sensor/bme68x_iaq/sample.yaml [hunk 6]
```diff
@@ -35,3 +35,11 @@ tests:
       ordered: true
       regex:
         - "(\\d+\\.\\d+); press: (\\d+\\.\\d+); humidity: (\\d+\\.\\d+); iaq: (\\d+)"
+  sample.sensor.bme68x.spi:
+    depends_on: spi
+    harness: sensor
+    integration_platforms:
```
Comment (maxd-nordic) [inline samples/sensor/bme68x_iaq/sample.yaml:41]: looks like we have to remove the `integration_platforms` part for the new test - my fault

**Comment by maxd-nordic** [comment]
could you rebase?

**Review by balaji-nordic — APPROVED** [review]

**Comment by purpl3F0x** [comment]
> could you rebase?

should I re-rebase now ? 😅

**Comment by maxd-nordic** [comment]
@purpl3F0x Thanks for your PR! It will be included in the next release. :)

**Pull request merged**